homeassistant/components/usb/__init__.py

```diff
@@ -243,26 +248,56 @@ def async_stop(self, event: Event) -> None:
         if self._request_debouncer:
             self._request_debouncer.async_shutdown()
 
-    async def _async_start_monitor(self) -> None:
-        """Start monitoring hardware with pyudev."""
-        if not sys.platform.startswith("linux"):
-            return
+    async def _async_supports_monitoring(self) -> bool:
         info = await system_info.async_get_system_info(self.hass)
         if info.get("docker"):
-            return
+            return False
+
+        return True
+
+    async def _async_start_monitor(self) -> None:
+        """Start monitoring hardware."""
+        if not await self._async_start_monitor_udev():
+            _LOGGER.info(
+                "Falling back to periodic filesystem polling for development, libudev "
+                "is not present"
+            )
+            await self._async_start_monitor_polling()
+
+    async def _async_start_monitor_polling(self) -> None:
```

Comment:
```suggestion
    @hass_callback
    def _async_start_monitor_polling(self) -> None:
```

Nothing awaited here